File: src/main/java/org/cobbzilla/util/cron/CronJob.java
package org.cobbzilla.util.cron;

import org.cobbzilla.util.reflect.ReflectionUtil;

import java.util.Properties;

public class CronJob {

    private String id;

    private String cronTimeString;

    private boolean startNow = false;

    private String commandClass;
    private Properties properties = new Properties();

    // todo
//    @Getter @Setter private String user;
//    @Getter @Setter private String shellCommand;

    public CronCommand getCommandInstance() {
        CronCommand command = ReflectionUtil.instantiate(commandClass);
        command.init(properties);
        return command;
    }

    public String getId() {
        return this.id;
    }

    public String getCronTimeString() {
        return this.cronTimeString;
    }

    public boolean isStartNow() {
        return this.startNow;
    }

    public String getCommandClass() {
        return this.commandClass;
    }

    public Properties getProperties() {
        return this.properties;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setCronTimeString(String cronTimeString) {
        this.cronTimeString = cronTimeString;
    }

    public void setStartNow(boolean startNow) {
        this.startNow = startNow;
    }

    public void setCommandClass(String commandClass) {
        this.commandClass = commandClass;
    }

    public void setProperties(Properties properties) {
        this.properties = properties;
    }
}

File: src/main/java/org/cobbzilla/util/cron/CronJob.kt
package org.cobbzilla.util.cron

import org.cobbzilla.util.reflect.ReflectionUtil

import java.util.Properties

class CronJob {

    var id: String? = null

    var cronTimeString: String? = null

    var isStartNow = false

    var commandClass: String? = null
    var properties = Properties()

    // todo
    //    @Getter @Setter private String user;
    //    @Getter @Setter private String shellCommand;

    val commandInstance: CronCommand
        get() {
            val command = ReflectionUtil.instantiate<CronCommand>(commandClass)
            command.init(properties)
            return command
        }
}
